src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -512,19 +526,25 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             {
                 sendTask = base.SendAsync(request, cts.Token);
             }
+            catch (TaskCanceledException e) when (TimeoutFired(timeoutState, cancellationToken))
```

Comment:
We should catch OperationCanceledException, not TaskCanceledException.  While we try to throw the latter for netfx compatibility, I expect there are some places where we don't, in particular if the cancellation exception originates from SslStream.

Reply:
If I understand correctly, HttpClient.Timeout is not passed down to the call chain and handled only on this level. Therefore, this logic handles only exceptions thrown by timeout-based cancellation triggered by the timer. All other exceptions are handled by generic catch block below the same way as they are now.